openlibrary/catalog/add_book/__init__.py

```diff
@@ -489,13 +489,18 @@ def find_quick_match(rec):
         if ekeys:
             return ekeys[0]
 
-    # only searches for the first value from these lists
+    # Look for a matching non-ISBN ASIN identifier (e.g. from a BWB promise item).
+    if (non_isbn_asin := get_non_isbn_asin(rec)) and (
```

Comment:
Not literally, but in spirit, `get_non_isbn_asin` is really:

`get_vendor_ids` -- returns a dict of `service` and `id` pairs which can be used to fetch/augment data (isbndb? amazon?)

e.g. {"amazon": "B0A8A8X"}

Reply:
The `service` variable may be aligned with e.g. "identifiers.amazon"